### DISCUSSION TITLE
Nvidia finally fixes the VRAM manager.

### DISCUSSION THREAD

**Discussion by berkut1** [oobabooga/text-generation-webui#4484] [General]
Just in case someone isn't keeping track of driver updates. In driver version 546.01+, the ability to disable the new memory manager has been added. 
I've tested it, and it works as intended without any performance loss as it was after 531+.
Just follow this guide - https://nvidia.custhelp.com/app/answers/detail/a_id/5490

**Comment by FartyPants** [comment]
Thanks for pointing to this. It should be probably added to docs. It's quite important!

Any difference between studio and game ready drivers?
also note, for WebUi if you use venv (one click install) you need to select the exe of python in vene, not in your system
Example:
N:\text-generation-webui-new\installer_files\env\python.exe

Here is chatgpt extract of that page, so it stays here as a copy:

Summary:
In driver version 536.40, NVIDIA introduced a new method to enable applications to use shared memory when GPU memory is exhausted, preventing crashes by allowing applications to run at lower speeds. However, Stable Diffusion, which often requires close to 6 GB of GPU memory, may invoke this mechanism on 6 GB GPUs, resulting in reduced application speed. Driver version 546.01 and above introduced a setting to disable the shared memory fallback, providing stable performance at the risk of potential crashes if GPU memory requirements are exceeded.

Instructions to Disable Shared Memory Fallback for Stable Diffusion:

    Run Stable Diffusion and note the path of the running python.exe process.
    Open NVIDIA Control Panel.
    In the 3D Settings section, click on Manage 3D settings.
    Navigate to the Program Settings tab and click Add to customize programs.
    Browse and select the Python executable file from Step 1.
    Click on CUDA - Sysmem Fallback Policy and set the value to Prefer No Sysmem Fallback.
    Click Apply to confirm.
    Re-launch Stable Diffusion for the changes to take effect.

**Reply by berkut1** [reply]
@FartyPants It usually takes 3-6 months for Studio to get a feature, so it will be coming soon too, or maybe they disable it by default in Studio, cause this feature useful only in games.
I'm just wondering how this will work on linux/docker/etc, maybe they will have to change some configs.

**Comment by simcop2387** [comment]
Anyone know if this affects the linux drivers and if it does how to disable it?

**Reply by berkut1** [reply]
You can easily check it. Load a big model with all GPU layout, if you not get the out of memory error, then linux has this issue too.

**Reply by navr32** [reply]
Some have a found a trick for the latest driver to overcome this problem on linux ?
